src/geometries/multigeom.jl

```diff
@@ -48,12 +48,14 @@ Base.isapprox(m₁::Multi, m₂::Multi; atol=atol(lentype(m₁)), kwargs...) =
 # POLYTOPE
 # ---------
 
-vertex(m::MultiPolytope, ind) = vertices(m)[ind]
+vertex(m::MultiPolytope, ind) = first(Iterators.drop(eachvertex(m), ind - 1))
 
-vertices(m::MultiPolytope) = [vertex for geom in m.geoms for vertex in vertices(geom)]
+vertices(m::MultiPolytope) = collect(eachvertex(m))
 
 nvertices(m::MultiPolytope) = sum(nvertices, m.geoms)
 
+eachvertex(m::MultiPolytope) = VertexItr(m)
```

Comment:
```suggestion
eachvertex(m::MultiPolytope) = (v for g in m.geoms for v in vertices(g))
```
The `VertexItr` is better for Multi with non-homogeneous geometries, but for homogeneous geometries, `Iterators.flatten` is better:
```julia
julia> itr1(m) = Meshes.VertexItr(m)
itr1 (generic function with 1 method)

julia> itr2(m) = (v for g in m.geoms for v in vertices(g))
itr2 (generic function with 1 method)

julia> function test1(m)
           s = 0
           for _ in itr1(m)
               s += 1
           end
           s
       end
test1 (generic function with 1 method)

julia> function test2(m)
           s = 0
           for _ in itr2(m)
               s += 1
           end
           s
       end
test2 (generic function with 1 method)

julia> r1 = rand(Ring, crs=Cartesian2D);

julia> r2 = rand(Ring, crs=Cartesian2D);

julia> r3 = rand(Ring, crs=Cartesian2D);

julia> r4 = rand(Ring, crs=Cartesian2D);

julia> p1 = PolyArea(r1, r2)
PolyArea
  outer
  └─ Ring((x: 0.679604 m, y: 0.060601 m), ..., (x: 0.0708917 m, y: 0.00982635 m))
  inner
  └─ Ring((x: 0.65877 m, y: 0.799253 m), ..., (x: 0.208459 m, y: 0.231627 m))

julia> p2 = PolyArea(r3, r4)
PolyArea
  outer
  └─ Ring((x: 0.792342 m, y: 0.685956 m), ..., (x: 0.0187149 m, y: 0.527211 m))
  inner
  └─ Ring((x: 0.509887 m, y: 0.190736 m), ..., (x: 0.346163 m, y: 0.540988 m))

julia> mp = Multi([p1, p2])
MultiPolyArea
├─ PolyArea(6-Ring, 16-Ring)
└─ PolyArea(21-Ring, 28-Ring)

julia> mr = Multi([r1, r2, r3, r4])
MultiRing
├─ Ring((x: 0.679604 m, y: 0.060601 m), ..., (x: 0.0708917 m, y: 0.00982635 m))
├─ Ring((x: 0.65877 m, y: 0.799253 m), ..., (x: 0.208459 m, y: 0.231627 m))
├─ Ring((x: 0.792342 m, y: 0.685956 m), ..., (x: 0.0187149 m, y: 0.527211 m))
└─ Ring((x: 0.509887 m, y: 0.190736 m), ..., (x: 0.346163 m, y: 0.540988 m))

julia> @btime test1($mp)
  23.461 μs (142 allocations: 49.86 KiB)
71

julia> @btime test1($mr)
  217.921 ns (0 allocations: 0 bytes)
71

julia> @btime test2($mp)
  578.714 ns (4 allocations: 1.25 KiB)
71

julia> @btime test2($mr)
  22.792 ns (0 allocations: 0 bytes)
```